### PR TITLE
Migrate to ml4gw preprocess

### PR DESCRIPTION
**Pull request by EthanMarx**
Migrates whitener and standard scaler to ml4gw versions

### CODE REVIEW

File: libs/trainer/tests/test_wrapper.py [hunk 1]
```diff
@@ -43,10 +42,10 @@ def to(self, device):
         return
 
 
-class Preprocessor(Transform):
+class Preprocessor(torch.nn.Module):
```
Comment (deepchatterjeeligo) [inline libs/trainer/tests/test_wrapper.py:45]: Wondering why this was a `Transform`

Comment (EthanMarx) [reply libs/trainer/tests/test_wrapper.py:45]: Don't think it needed to be. If you're curious, the concept has been moved to `ml4gw` and is now called `FittableTransform`

File: projects/sandbox/train/train.py [hunk 2]
```diff
@@ -37,7 +40,7 @@ def load_signals(waveform_dataset: Path, parameter_names: List[str]):
                     values = np.log10(values)
                 data.append(values)
 
-        intrinsic = np.column_stack(data)
+        intrinsic = np.row_stack(data)
```
Comment (deepchatterjeeligo) [inline projects/sandbox/train/train.py:43]: Was this a bug previously?

Comment (EthanMarx) [reply projects/sandbox/train/train.py:43]: Not a bug: This is a result of moving to `ChannelWiseScaler` in ml4gw (see comment in `preprocessor.py`)

File: projects/sandbox/train/train.py [hunk 3]
```diff
@@ -134,6 +130,8 @@ def main(
     preprocessor.normalizer.fit(intrinsic)
     preprocessor.normalizer.to(device)
 
+    torch.save(preprocessor, outdir / "preprocessor.pt")
```
Comment (deepchatterjeeligo) [inline projects/sandbox/train/train.py:133]: How do we re-use this?

File: libs/data/mlpe/data/transforms/preprocessor.py [hunk 4]
```diff
@@ -17,32 +15,43 @@ class Preprocessor(torch.nn.Module):
...
 
     def __init__(
         self,
         num_ifos: int,
         sample_rate: float,
-        kernel_length: float,
-        normalizer: Optional["Transform"] = None,
         fduration: Optional[float] = None,
-        highpass: Optional[float] = None,
+        normalizer: torch.nn.Module = torch.nn.Identity(),
```
Comment (deepchatterjeeligo) [inline libs/data/mlpe/data/transforms/preprocessor.py:36]: Why is the normalizer default identity?

File: projects/sandbox/train/train.py (hunk 3 above)
Comment (EthanMarx) [reply projects/sandbox/train/train.py:133]: I *think* we'll be able to call preprocessor.load_state_dict(). Should double check that

File: libs/data/mlpe/data/transforms/preprocessor.py (hunk 4 above)
Comment (EthanMarx) [reply libs/data/mlpe/data/transforms/preprocessor.py:36]: So that it will just return the parameters themselves if a normalizer isn't passed

Comment (deepchatterjeeligo) [reply libs/data/mlpe/data/transforms/preprocessor.py:36]: Minor nitpick. But can we call it something different. Normalizer sounds like it will always normalize. Maybe `scaler`?

Comment (EthanMarx) [reply libs/data/mlpe/data/transforms/preprocessor.py:36]: Sure

File: libs/trainer/pyproject.toml [hunk 5]
```diff
@@ -8,16 +8,25 @@ packages = [{include = "mlpe"}]
 
 [tool.poetry.dependencies]
 python = ">=3.8,<3.11"
+torch = {version = "^1.12", source = "torch"}
```
Comment (deepchatterjeeligo) [inline libs/trainer/pyproject.toml:11]: Just a quick check, as a part of this maintenance patch. Shall we remove the minor tick constraint. I noticed some of the projects/libs were installing 1.13 and others were install 1.12. It's not a problem, but I am wondering if we should just do
```suggestion
torch = {version = ">=1.12", source = "torch"}
```

File: libs/trainer/tests/test_wrapper.py (hunk 1 above)
Comment (deepchatterjeeligo) [reply libs/trainer/tests/test_wrapper.py:45]: Fittable transform seem like a better option here, with a `build` function etc. no?

**Comment by deepchatterjeeligo** [comment]
This looks good. I'll approve it now. Feel free to merge when you have some of the minor details above done.

**Review by deepchatterjeeligo — APPROVED** [review]

File: projects/sandbox/train/train.py (hunk 3 above)
Comment (EthanMarx) [reply projects/sandbox/train/train.py]: Fixed this so that we save the `whitener` and `scaler` individually

File: libs/trainer/pyproject.toml (hunk 5 above)
Comment (EthanMarx) [reply libs/trainer/pyproject.toml:11]: Remind me: what is the exact difference here? I agree should definitely standardize versions
